src/main/scala/org/broadinstitute/hail/stats/FisherExactTest.scala

```diff
@@ -0,0 +1,270 @@
+package org.broadinstitute.hail.stats
+
+import org.apache.commons.math3.distribution.HypergeometricDistribution
+import org.broadinstitute.hail.Utils.{fatal, divOption}
+
+object FisherUtils {
+  //based on C code in R source code called zeroin.c
+  def uniroot(fn: Double => Double, min: Double, max: Double, tolerance: Double = 1.220703e-4): Option[Double] = {
```

Comment:
Root finding is generally useful.  I'd put this ... in the stats package object, maybe?  You might consider making FisherExactTest into a function in the package object, also, since you don't use its objectness.